rkt/gc.go

```diff
@@ -188,50 +235,13 @@ func deletePod(p *pod) {
 	}
 
 	if p.isExitedGarbage {
-		s, err := store.NewStore(globalFlags.Dir)
-		if err != nil {
-			stderr("Cannot open store: %v", err)
-			return
-		}
-		defer s.Close()
-		stage1TreeStoreID, err := p.getStage1TreeStoreID()
-		if err != nil {
-			stderr("Error getting stage1 treeStoreID: %v", err)
-			return
-		}
-		stage1RootFS := s.GetTreeStoreRootFS(stage1TreeStoreID)
-
-		// execute stage1's GC
-		if err := stage0.GC(p.path(), p.uuid, stage1RootFS, globalFlags.Debug); err != nil {
-			stderr("Stage1 GC of pod %q failed: %v", p.uuid, err)
-			return
-		}
-
-		if p.usesOverlay() {
-			apps, err := p.getApps()
-			if err != nil {
-				stderr("Error retrieving app hashes from pod manifest: %v", err)
-				return
+		if err := cleanExitedGarbage(p); err != nil {
```

Comment:
I don't know if this seems right. This goes straight to the force unmounting irrespective of the failure condition, e.g. if https://github.com/coreos/rkt/pull/1828/files#diff-204288dd8b5d5205881366c76eb4f97eR184 fails initialising the store. Might that not cause pods to get in a weird half-GCed state? Should we be more nuanced at distinguishing actual failure conditions here?